plugins/outputs/zabbix/lld_test.go

```diff
@@ -0,0 +1,1234 @@
+						1: {
+							"{#FOO}": "bar",
+						},
+					},
+				},
+			},
+			PreviousReceivedData: map[uint64]lldInfo{
+				2658406801034663970: {
+					Hostname: "hostA",
+					Key:      "disk.foo",
+					Data: map[uint64]map[string]string{
+						1: {
+							"{#FOO}": "bar",
+						},
+					},
+				},
+			},
+			Metrics: []telegraf.Metric{},
+		},
+		"send an empty LLD if one metric has stopped being sent": {
+			ReceivedData: map[uint64]lldInfo{},
+			PreviousReceivedData: map[uint64]lldInfo{
+				0: {
+					Hostname: "hostA",
+					Key:      "disk.foo",
+					Data: map[uint64]map[string]string{
+						1: {
+							"{#FOO}": "bar",
+						},
+					},
+				},
+			},
+			Metrics: []telegraf.Metric{
+				testutil.MustMetric(lldName,
+					map[string]string{"host": "hostA"},
+					map[string]interface{}{"disk.foo": `{"data":[]}`},
+					time.Now(),
+				),
+			},
+		},
+	}
+
+	for desc, test := range tests {
+		t.Run(desc, func(t *testing.T) {
+			zl := zabbixLLD{
+				clearInterval: 4,
+				log:           testutil.Logger{},
+				current:       test.ReceivedData,
+				previous:      test.PreviousReceivedData,
+				hostTag:       "host",
+			}
+
+			// Hash the previous data
+			for series, info := range zl.previous {
+				info.DataHash = info.hash()
+				zl.previous[series] = info
+			}
+
+			metrics := zl.Push()
+			// Sort the "data" dict in the metrics values to get always the same order.
+			metrics = sortMetricJSONData(metrics)
+
+			testutil.RequireMetricsEqual(t, test.Metrics, metrics, testutil.IgnoreTime(), testutil.SortMetrics())
+		})
+	}
+}
+
+type MetricValue struct {
+	Data []map[string]string `json:"data"`
+}
+
+// sortMetricJSONData given a list of metrics, if the name is equal to lldName, the JSON data dictionaries are sorted.
+// This is needed because the order of the JSON data dictionaries is not guaranteed but we need them sorted to compare
+// them against the expected tests values. This sorting should be done using the keys of the dictionaries.
+// Example:
+//
+//	Original metrics: lld,host=foo disk.foo={"data":[{"{#FOO2}":"bar2"},{"{#FOO1}":"bar1"}]}
+//	Sorted metrics:   lld,host=foo disk.foo={"data":[{"{#FOO1}":"bar1"},{"{#FOO2}":"bar2"}]}
+func sortMetricJSONData(metrics []telegraf.Metric) []telegraf.Metric {
+	for _, m := range metrics {
+		if m.Name() == lldName {
+			for _, f := range m.FieldList() {
+				// f is a string with format: '{"data":[{"{#PID}":"1234"}]}'
+				var data MetricValue
+				err := json.Unmarshal([]byte(f.Value.(string)), &data)
+				if err != nil {
+					panic(err)
+				}
+
+				// Sort data comparing the content as a string
+				sort.Slice(data.Data, func(i, j int) bool {
+					return fmt.Sprintf("%v", data.Data[i]) < fmt.Sprintf("%v", data.Data[j])
+				})
+
+				dataJSON, err := json.Marshal(data)
+				if err != nil {
+					panic(err)
+				}
+
+				f.Value = string(dataJSON)
+			}
+		}
+	}
+
+	return metrics
+}
+
+/*
+func TestCompareAndDelete(t *testing.T) {
```

Comment:
Why are the remaining tests commented out?

Reply:
My bad. I commented that part while adapting to the srebhan version. I have to update those tests.

Reply:
Fixed in 0b1c4528211b995f76744ddafdcc54efab73831e